### PR TITLE
Sync upstream v0.6.2 next

### PR DESCRIPTION
**Pull request by debaghtk**
## How this was tested
this was tested locally for a single order and 5 nodes

### CODE REVIEW

**Comment by debaghtk** [comment]
@shubhamgoyal42 question, is chain.json for local env only? and do i need to add my additional config in aylin as well as mainnet?

**Comment by shubhamgoyal42** [comment]
> @shubhamgoyal42 question, is chain.json for local env only? and do i need to add my additional config in aylin as well as mainnet?

yes, chain.json is only for local. You'll need to add the respective configs in network-configs dir

File: plugin/evm/config.go [hunk 1]
```diff
@@ -173,6 +173,11 @@ type Config struct {
 	RegossipFrequency         Duration         `json:"regossip-frequency"`
 	PriorityRegossipAddresses []common.Address `json:"priority-regossip-addresses"`
 
+	// Order Gossip Settings
+	OrderGossipNumValidators    int `json:"order-gossip-num-validators"`
```
Comment (shubhamgoyal42) [inline plugin/evm/config.go:177]: should we add defaults in code, like other config fields?

File: plugin/evm/gossiper_orders.go [hunk 2]
```diff
@@ -2,25 +2,100 @@ package evm
...
+	lastOrdersGossiped time.Time
+
+	codec  codec.Manager
+	signer types.Signer
+	stats  GossipStats
+
+	appSender commonEng.AppSender
+}
+
+// createGossiper constructs and returns a pushGossiper or noopGossiper
```
Comment (shubhamgoyal42) [inline plugin/evm/gossiper_orders.go:65]: 1. nit: ~createGossiper~ createOrderGossiper
2. second line can be removed

File: plugin/evm/gossiper_orders.go [hunk 3]
```diff
@@ -2,25 +2,100 @@ package evm
...
+	GossipSignedOrders(orders []*hubbleutils.SignedOrder) error
+}
+
+type orderPushGossiper struct {
+	ctx    *snow.Context
+	config Config
+
+	client     peer.NetworkClient
+	blockchain *core.BlockChain
+	txPool     *txpool.TxPool
```
Comment (shubhamgoyal42) [inline plugin/evm/gossiper_orders.go:49]: these 3 are not being used anywhere

File: plugin/evm/gossiper_orders.go [hunk 4]
```diff
@@ -2,25 +2,100 @@ package evm
...
+
+	shutdownChan chan struct{}
+	shutdownWg   *sync.WaitGroup
+
+	ordersToGossipChan chan []*hubbleutils.SignedOrder
+	ordersToGossip     []*hubbleutils.SignedOrder
+	lastOrdersGossiped time.Time
+
+	codec  codec.Manager
+	signer types.Signer
```
Comment (shubhamgoyal42) [inline plugin/evm/gossiper_orders.go:59]: this one too

Comment (debaghtk) [reply plugin/evm/gossiper_orders.go]: done

File: plugin/evm/config.go (hunk 1 above)
Comment (debaghtk) [reply plugin/evm/config.go:180]: done

**Review by shubhamgoyal42 — APPROVED** [review]

**Pull request merged**